middle_end/flambda/un_anf.ml

```diff
@@ -172,20 +195,28 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
       loop ~depth ~environment_vars body;
       environment_vars
     | Uletrec (defs, body) ->
+    let environment_vars =
       List.fold_left (fun (var, def) ->
           ignore_var_with_provenance var;
           loop ~depth ~environment_vars def)
-        defs;
-      loop ~depth ~environment_vars body;
       environment_vars
+      defs
+    in
+    loop ~depth ~environment_vars body;
+    environment_vars
     | Uprim (prim, args, dbg) ->
       ignore_primitive prim;
-      List.fold_left (loop ~depth ~environment_vars) args;
+      let environment_vars =
+        List.fold_left
+          (fun environment_vars arg -> loop ~depth ~environment_vars arg)
+          environment_vars
+          args
+      in
       ignore_debuginfo dbg;
       environment_vars
     | Uswitch (cond, { us_index_consts; us_actions_consts;
           us_index_blocks; us_actions_blocks }, dbg) ->
-      loop ~depth ~environment_vars cond;
+      let environment_vars = loop ~depth ~environment_vars cond in
       ignore_int_array us_index_consts;
       Array.iter (loop ~depth ~environment_vars) us_actions_consts;
```

Comment:
The `Array` module also exports a `fold_left` function, so you can do the same transformation her as for lists

Reply:
(And two lines below, too)